### DISCUSSION TITLE
Extra wide side modal

### DISCUSSION THREAD

**Discussion by darcimm** [twilio-labs/paste#3167] [General] [answered]
Is it possible to create an additional wider side modal for a journey analytics use case?

**Context**
In a published journey, each step opens as a side sheet to see the conditions that were set for that step:
![image](https://user-images.githubusercontent.com/83974127/232581438-7a6e15ee-5f16-443e-b64a-5e81d98d0441.png)

On an email, SMS, or Whatsapp step, the side sheet currently includes (some) engagement analytics:
![image](https://user-images.githubusercontent.com/83974127/232581517-c10ea1c7-25cc-41d0-8346-da21a5da0114.png)

**Problem**
As we add more analytics/visualizations to the journey side sheet, there are two problems we're trying to solve:
1) More space for additional visualizations
2) More alignment with Single Sends designs

Here's what the Single Sends analytics engagement metrics looks like:
![image](https://user-images.githubusercontent.com/83974127/232581981-24355ee6-2245-430c-93b8-3d19dc848a55.png)

**Proposed Solution**
Here's what we'd like to do in the journey side sheet to better align with Single Sends (and add more visualizations in a 'dashboard' fashion):
![image](https://user-images.githubusercontent.com/83974127/232582093-0ffaccca-870d-4ca2-b9bd-07a7798d7860.png)
- The design is currently fixed width at ~1152px, or we could consider an 80% width. Open to options!

There are still some moving parts but wanted to get this in front of y'all sooner rather than later. Looking forward to your thoughts. Thanks!

**Comment by loreina** [comment, marked as answer]
Hi @darcimm,

Spoke with the team and we aren't keen on adding a variant to Side Modal at 80% width as the rest of the page is still interactive and it would cover nearly the entire content area aside from the sidebar navigation.

Instead, we'd recommend going with one of these solutions, in this order:
1. Using [Side Modal](https://paste.twilio.design/components/side-modal) and sticking with the constraints of the current width.
2. Building this component using our [Modal Dialog Primitive](https://paste.twilio.design/primitives/modal-dialog-primitive). Feel free to drop by office hours with your engineer if you need any guidance on matching Side Modal styles with the primitive.

Thanks!
